src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -159,18 +173,23 @@ internal override GeneratedLabelSymbol BreakLabel
 
         protected override ImmutableArray<LabelSymbol> BuildLabels()
         {
-            ArrayBuilder<LabelSymbol> labels = null;
+            // We bind the switch expression and the switch case label expressions so that the constant values can be
+            // part of the label, but we do not report any diagnostics here. Diagnostics will be reported during binding.
 
+            ArrayBuilder<LabelSymbol> labels = ArrayBuilder<LabelSymbol>.GetInstance();
+            DiagnosticBag tempDiagnosticBag = DiagnosticBag.GetInstance();
+            TypeSymbol switchGoverningType = this.SwitchGoverningType;
             foreach (var section in _switchSyntax.Sections)
             {
                 // add switch case/default labels
-                BuildSwitchLabels(section.Labels, GetBinder(section), ref labels);
+                BuildSwitchLabels(switchGoverningType, section.Labels, GetBinder(section), labels, tempDiagnosticBag);
 
                 // add regular labels from the statements in the switch section
-                base.BuildLabels(section.Statements, ref labels);
+                BuildLabels(section.Statements, ref labels);
```

Comment:
I don't think there's any use in having `ref labels` anymore if you're going to eagerly create the builder.